File: Scripts/file_name_changer.py
import subprocess as subprocess
import os
import platform

opSystem = platform.system()

if(opSystem == 'Linux'):

	pathBase = os.path.split(os.path.dirname(os.path.abspath(__file__)))[0] + '/'
	pathSufix = 'Datasets/Dataset_Bovespa/'
	pathDestiny = 'Unzipped/'
	
	location = pathBase + pathSufix + pathSender

	filePrefix = 'COTAHIST.A'
	fileStartYear = 1986
	fileEndYear = 2000

    filePrefixDesired = "COTAHIST_A"
    fileExtension = '.txt'

	linuxCommand = 'mv'

	for year in range(fileStartYear, fileEndYear + 1):
        completeCommand = linuxCommand + ' ' + location + filePrefix + str(year) + ' ' + filePrefixDesired + str(year) + fileExtension
        print(completeCommand)
        result = subprocess.Popen(completeCommand, shell=True)
        print(result)
    print("Conversion done from 1986 to 1999.")
    
    completeCommand = powerShellCommand + location + filePrefixDesired + str(2000) + ' ' + commandComplement + ' ' + filePrefixDesired + str(2000) + fileExtension
    result = subprocess.Popen(completeCommand, shell=True)
    print(result)

if(opSystem == 'Windows'):

    pathBase = os.path.split(os.path.dirname(os.path.abspath(__file__)))[0] + '\\'

    pathSufix = 'Datasets\\Dataset_Bovespa\\'
    pathSender = 'Unzipped\\'

    location = pathBase + pathSufix + pathSender

    filePrefix = 'COTAHIST.A'
    fileStartYear = 1986
    fileEndYear = 2000

    filePrefixDesired = 'COTAHIST_A'
    fileExtension = '.txt'

    powerShellCommand = 'Rename-Item -Path '
    commandComplement = '-NewName'

    for year in range(fileStartYear, fileEndYear + 1):
        completeCommand = powerShellCommand + location + filePrefix + str(year) + ' ' + commandComplement + ' ' + filePrefixDesired + str(year) + fileExtension
        print(completeCommand)
        result = subprocess.Popen(['powershell.exe',completeCommand])
        print(result)
    print("Conversion done from 1986 to 1999.")
    
    completeCommand = powerShellCommand + location + filePrefixDesired + str(2000) + ' ' + commandComplement + ' ' + filePrefixDesired + str(2000) + fileExtension
    result = subprocess.Popen(['powershell.exe',completeCommand])
    print(result)

File: Scripts/file_name_changer.jl
using Base.Filesystem
const fs = Base.Filesystem

path_base = dirname(pwd())
path_sufix = "/Datasets/Dataset_Bovespa/"
path_destiny = "Unzipped/"

location = path_base * path_sufix * path_destiny

file_prefix = "COTAHIST.A"
file_start_year = 1995
file_end_year = 2000

file_prefix_desired = "COTAHIST_A"
file_extension = ".TXT"

for year in range(file_start_year, stop=file_end_year)
    path_original = location * file_prefix * string(year)
    path_renamed = location * file_prefix_desired * string(year) * file_extension
    fs.mv(path_original, path_renamed)
end

path_original = location * file_prefix_desired * "2001"
path_renamed = location * file_prefix_desired * "2001" * file_extension
fs.mv(path_original, path_renamed)

println("Files from 1995 to 2001 were successfully converted!!")